Add STM32 L452 with XC soc variant

GeorgeCGV opened the pull request:
Adds base STM32 L452 soc with L452XC variant. 
L452 has 160 KB SRAM; therefore, mpu mem config has been extended.

erwango commented on dts/arm/st/l4/stm32l452.dtsi:
```diff
@@ -0,0 +1,136 @@
+ */
+
+#include <mem.h>
+#include <st/l4/stm32l4.dtsi>
```
You should be able to include another l4 .dtsi such as stm32l433 for instance

GeorgeCGV replied:
I see, now it shall be the right way

erwango commented on dts/arm/st/l4/stm32l452.dtsi:
```diff
@@ -0,0 +1,136 @@
+#include <st/l4/stm32l4.dtsi>
+
+/ {
+	sram0: memory@20000000 {
```
Put this in package .dtsi

GeorgeCGV replied:
Moved the part to `xC.dtsi`

erwango reviewed (changes requested):
Thanks for the change.
First comments after initial review

GeorgeCGV commented:
@erwango I moved the memory def. into `xC.dtsi` and made L452 to extend L433, missed that initially

erwango reviewed (commented):
@GeorgeCGV , I had a closer look and seems you were right. 
STM32L432/3 is not a subset of STM32L452. STM32L432 has TIM7 and SWPMI1 which are not on STM32L452. So you could actually include stm32l4.dtsi.
Then also please squash your commits.

Commit:
soc: st: add STM32L452XC

Adds STM32 L452 soc with L452xC variant. L452 has 160 KB SRAM;
therefore, mpu mem config has been extended; IRQ number is based on
'stm32l452xx.h'.

Signed-off-by: Georgij Cernysiov <g.cernysiov@elco-automation.de>

GeorgeCGV commented on soc/arm/st_stm32/stm32l4/Kconfig.defconfig.stm32l452xx:
```diff
@@ -0,0 +1,32 @@
+
+config NUM_IRQS
+	int
+	default 83
```
As discussed,

the reference manual mentions 
>  67 maskable interrupt channels plus the 16 interrupt lines

however, CMSIS `stm32l452xx.h` defines `85`; will switch to the CMSIS variant.

In order to not use nodes deletion I revert L452 to the previous state. Memory def. stays in the package dtsi.

GeorgeCGV commented:
> @erwango I thought it was pretty much a requirement to not merge a new SOC without an accompanying board. Has this notion changed in zephyr?

@psidhu, I don't have nucleo-l452re at hands. Currently, I am using l452rc on a custom board. 
Can add official nucleo board as soon as I have some spare time.

erwango commented:
@psidhu, you're right this was indeed the case some time ago but following some discussions we have relaxed this constraint so people can develop their product w/o the constraint of upstreaming it. Of course this is doable only if the new SoC does not require new code introduction and is only the addition of dtsi and few Kconfig options.
This being said, it seems that out of tree SoC support is now available and we might reconsider this position.
@GeorgeCGV, nucleo-l452re addition would be welcomed, but please do it only if you have it at hands.

GeorgeCGV commented:
@erwango ok, but I don't expect it to be around quite soon. Maybe someone else can extend this pull request with the board at hands. Usage of out of tree soc is not a bad option as well.

> ...not merge a new SOC without an accompanying board...

Would be great if such information (_e.g. what kind of pull requests are considered for acceptance in relation to what they bring_) was in the [Contribution Guidelines](https://docs.zephyrproject.org/latest/contribute/index.html).
